drivers/watchdog/wdt_ite_it8xxx2.c

```diff
@@ -0,0 +1,258 @@
+	 */
+	inst->ETWCFG = (EWDKEYEN | EWDSRC | LEWDCNTL | LET1PS | LETWCFG);
+
+	LOG_DBG("WDT Setup and enabled");
+
+	return 0;
+}
+
+static int wdt_it8xxx2_feed(const struct device *dev, int channel_id)
+{
+	struct wdt_it8xxx2_data *data = DRV_DATA(dev);
+	struct wdt_it8xxx2_regs *const inst = DRV_REG(dev);
+	uint16_t cnt0 = WARNING_TIMER_PERIOD_MS_TO_1024HZ_COUNT(data->timeout);
+
+	ARG_UNUSED(channel_id);
+
+	/* reset pre-warning timer1 */
+	inst->ETWCTRL |= ET1RST;
+
+	/* restart watchdog timer */
+	inst->EWDKEYR = IT8XXX2_WATCHDOG_MAGIC_WORD;
+
+	/* reset pre-warning timer1 to default if time is touched */
+	if (wdt_warning_fired) {
+		wdt_warning_fired = 0;
+
+		/* pre-warning timer1 is 16-bit counter down timer */
+		inst->ET1CNTLHR = (cnt0 >> 8) & 0xff;
+		inst->ET1CNTLLR = cnt0 & 0xff;
+
+		/* clear timer1 interrupt status */
+		ite_intc_isr_clear(DT_INST_IRQN(0));
+
+		/* enable timer1 interrupt */
+		irq_enable(DT_INST_IRQN(0));
+	}
+
+	LOG_DBG("WDT Kicking");
+
+	return 0;
+}
+
+static int wdt_it8xxx2_disable(const struct device *dev)
+{
+	struct wdt_it8xxx2_data *data = DRV_DATA(dev);
+	struct wdt_it8xxx2_regs *const inst = DRV_REG(dev);
+
+	/* stop watchdog timer counting */
+	inst->ETWCTRL |= EWDSCEN;
+
+	/* disable pre-warning timer1 interrupt */
+	irq_disable(DT_INST_IRQN(0));
+
+	/* mark uninstalled */
+	data->timeout_installed = false;
+
+	LOG_DBG("WDT Disabled");
+
+	return 0;
+}
+
+static void wdt_it8xxx2_isr(const struct device *dev)
+{
+	struct wdt_it8xxx2_data *data = DRV_DATA(dev);
+	struct wdt_it8xxx2_regs *const inst = DRV_REG(dev);
+	uint16_t cnt0 = WARNING_TIMER_PERIOD_MS_TO_1024HZ_COUNT(data->timeout);
+
+	/* clear pre-warning timer1 interrupt status */
+	ite_intc_isr_clear(DT_INST_IRQN(0));
+
+	/* reset pre-warning timer1 */
+	inst->ETWCTRL |= ET1RST;
+
+	/* callback function, ex. print warning message */
+	if (data->cb) {
+		data->cb(dev, 0);
+	}
+
+	/*
+	 * Reduce interval of warning timer, so we can print more
+	 * warning messages during critical period.
+	 */
+	if (!wdt_warning_fired++) {
+		/* pre-warning timer1 is 16-bit counter down timer */
+		inst->ET1CNTLHR = (cnt0 >> 8) & 0xff;
+		inst->ET1CNTLLR = cnt0 & 0xff;
+
+		/* clear pre-warning timer1 interrupt status */
+		ite_intc_isr_clear(DT_INST_IRQN(0));
+	}
+
+	LOG_DBG("WDT ISR");
+}
+
+static const struct wdt_driver_api wdt_it8xxx2_api = {
+	.setup = wdt_it8xxx2_setup,
+	.disable = wdt_it8xxx2_disable,
+	.install_timeout = wdt_it8xxx2_install_timeout,
+	.feed = wdt_it8xxx2_feed,
+};
+
+static int wdt_it8xxx2_init(const struct device *dev)
+{
+	struct wdt_it8xxx2_regs *const inst = DRV_REG(dev);
+
+	if (IS_ENABLED(CONFIG_WDT_DISABLE_AT_BOOT)) {
+		wdt_it8xxx2_disable(dev);
+	}
+
+	/* unlock access to watchdog registers */
+	inst->ETWCFG = 0x00;
+	/* set WDT and timer1 to use 1.024kHz clock */
+	inst->ET1PSR = 0x01;
+	/* set WDT key match enabled and WDT clock to use ET1PSR */
+	inst->ETWCFG = 0x30;
+	/* watchdog can be stopped */
```

Comment:
defines for the above three magic numbers would make the code more readable.

Reply:
Done